packages/cli/src/cmds/agentInstaller/pythonAgentInstaller.ts

```diff
@@ -95,6 +95,54 @@ export class PoetryInstaller extends PythonInstaller {
   }
 }
 
+export class PipenvInstaller extends PythonInstaller {
+  constructor(path: string) {
+    super('pipenv', path);
+  }
+
+  get buildFile(): string {
+    return 'Pipfile.lock';
+  }
+
+  get buildFilePath(): string {
+    return join(this.path, this.buildFile);
+  }
+
+  async available(): Promise<boolean> {
+    const ret = await exists(this.buildFilePath);
+    return ret;
+  }
+
+  async initCommand(): Promise<CommandStruct> {
+    return new CommandStruct('pipenv', ['run', 'appmap-agent-init'], this.path);
+  }
+
+  async checkCurrentConfig(): Promise<void> {
+    // pipenv doesn't seem to offer a way to check the config, short of doing an install.
```

Comment:
So, should we do an install then? I think that's how this feature should work.

Reply:
Yeah, that's what I do for `yarn` because there isn't a dry run feature

Reply:
Let me see what happens when I do that. If both installs take the same amount of time (i.e. this doubles the time to install the agent), does it still make sense to do this?

Reply:
@kgilpin I confirmed that doing this does take twice as long (if the user hasn't previously installed the dependencies).

Should I still make this change?